examples/ex25.cpp

```diff
@@ -388,32 +408,35 @@ int main(int argc, char *argv[])
    a.AddDomainIntegrator(new VectorFEMassIntegrator(restr_c2_Re),
                          new VectorFEMassIntegrator(restr_c2_Im));
 
-   // 12. Assemble the bilinear form and the corresponding linear system,
+   // 13. Assemble the bilinear form and the corresponding linear system,
    //     applying any necessary transformations such as: assembly, eliminating
    //     boundary conditions, applying conforming constraints for
    //     non-conforming AMR, etc.
+   if (pa) { a.SetAssemblyLevel(AssemblyLevel::PARTIAL); }
    a.Assemble(0);
 
    OperatorPtr A;
    Vector B, X;
    a.FormLinearSystem(ess_tdof_list, x, b, A, X, B);
 
-   // 13. Solve using a direct or an iterative solver
+   // 14. Solve using a direct or an iterative solver
 #ifdef MFEM_USE_SUITESPARSE
+   if (!pa && umf_solver)
    {
       ComplexUMFPackSolver csolver(*A.As<ComplexSparseMatrix>());
       csolver.Control[UMFPACK_ORDERING] = UMFPACK_ORDERING_METIS;
       csolver.SetPrintLevel(1);
       csolver.Mult(B, X);
    }
-#else
-   // 13a. Set up the Bilinear form a(.,.) for the preconditioner
+#endif
+   // 14a. Set up the Bilinear form a(.,.) for the preconditioner
    //
    //    In Comp
    //    Domain:   1/mu (Curl E, Curl F) + omega^2 * epsilon (E,F)
    //
    //    In PML:   1/mu (abs(1/det(J) J^T J) Curl E, Curl F)
    //              + omega^2 * epsilon (abs(det(J) * (J^T J)^-1) * E, F)
+   if (pa || !umf_solver)
```

Comment:
I think it would be better to choose one of the two if both `-pa` and `-umf_solver` are true, i.e., something like ex1. Similarly for ex25p.

Reply:
It seems there has been deadlock with @stefanhenneking waiting on us reviewers to choose a resolution to this issue. Currently, the case without a solve is if `MFEM_USE_SUITESPARSE` is not defined and `!pa && umf_solver`. I think in this case there should be an error. @psocratis mentioned the case of `pa && umf_solver`, which currently is solved by GMRES. That is a separate issue, and I am fine with the current implementation since one solver is chosen.

Reply:
> It seems there has been deadlock with @stefanhenneking waiting on us reviewers to choose a resolution to this issue. Currently, the case without a solve is if `MFEM_USE_SUITESPARSE` is not defined and `!pa && umf_solver`. I think in this case there should be an error. @psocratis mentioned the case of `pa && umf_solver`, which currently is solved by GMRES. That is a separate issue, and I am fine with the current implementation since one solver is chosen.

Hey @dylan-copeland 

if `MFEM_USE_SUITESPARSE` is not defined then there is no option for the user to set `umf_solver` which is by default `false`. So it falls back to the case `(pa || !umf_solver)` and since it's `!pa` it uses the `GSSmoother` as a preconditioner.

Reply:
Thanks @psocratis for pointing this out. My comment was just based on confusion.